new-pipeline/dist.html

```diff
@@ -55,7 +55,11 @@ <h2 style="font-size: 18px; margin-bottom: 0">
                 </h2>
                 <hr></hr>
                 <figure class="col-md-12">
-                  <figcaption id="dist-caption"></figcaption>
+                  <figcaption id="dist-caption" class="overflow ellipsis"></figcaption>
+                  <div class="caption-div">
+                  <figcaption id="caption-dots" >.....</figcaption>
```

Comment:
I think this is not working right yet. The "..." should be added automatically through the "text-overflow: ellipsis;" command.
Let's maybe try this isolated without all the existing code around first. I think [this is](https://jsfiddle.net/exzLLez0/11/) roughly the structure that we want here (not pretty, but shows the idea).

Can we apply this idea in this dashboard?

Reply:
@georgf ,  I had to add "..." separately because If we put the content(caption) directly into the `HTML` tag `<figcaption></figcaption>`, then `"..."` comes automatically.
But if the caption is rendered using JavaScript, then this `"..."` doesn't come.
Otherwise, the shortening of caption is taking place as per the desired UI.

Please see the screenshot attached.

![screenshot from 2018-03-06 02-11-45](https://user-images.githubusercontent.com/23060217/36998867-6293ff02-20e4-11e8-9fdf-1cfbfd8ec933.png)
![screenshot from 2018-03-06 02-11-52](https://user-images.githubusercontent.com/23060217/36998868-62c77d1e-20e4-11e8-9c09-e32a9888883f.png)

Any advice on what and where to get the changes done is most welcome! 
Thanks!

Reply:
I think the ellipsis in the screenshot looks fine. The only issues is that the "more details" button is on a new line.
Am i missing some problem here? :)

Reply:
The `ellipsis` looks fine in the screen-shot because I have put an example caption in the `HTML`tag directly.  Please see the code in the screen-shot.

![screenshot from 2018-03-06 02-11-45](https://user-images.githubusercontent.com/23060217/37000442-6c97e2ca-20e9-11e8-88f0-e7a4ad91f310.png)

This code shows up like : 

![screenshot from 2018-03-06 02-11-52](https://user-images.githubusercontent.com/23060217/37000758-596280ce-20ea-11e8-9a28-f52a092e6690.png)

The problem comes when the caption is rendered using JavaScript in the runtime from `getDescription` function here [https://github.com/mozilla/telemetry-dashboard/blob/gh-pages/new-pipeline/src/dashboards.js#L1059](url) .
In this case, the caption does shorten up like what a normal `ellipsis` does except `"..."` doesn't come. Hence, I had to add `"..."` separately.

![screenshot from 2018-03-06 02-57-23](https://user-images.githubusercontent.com/23060217/37000767-6112a524-20ea-11e8-9541-f2e1c820da3b.png)

This shows up like :

![screenshot from 2018-03-06 03-00-37](https://user-images.githubusercontent.com/23060217/37000779-6d8bc132-20ea-11e8-9625-f370f97df061.png)

Does this sound clear?

Reply:
@georgf , I have found a fix for my problem. I will report back in a while. Thanks!